hooks/useUpdateCENNZBalances.ts

```diff
@@ -0,0 +1,29 @@
+import { Dispatch, SetStateAction, useCallback } from "react";
+import fetchCENNZAssetBalances from "@/utils/fetchCENNZAssetBalances";
+import { useCENNZApi } from "@/providers/CENNZApiProvider";
+import { useWalletProvider } from "@/providers/WalletProvider";
+import { useSelectedAccount } from "@/hooks";
+import { CENNZAssetBalance } from "@/types";
+
+export default function useUpdateCENNZBalances(): (
+	setBalances: Dispatch<SetStateAction<CENNZAssetBalance[]>>
+) => Promise<void> {
+	const { api } = useCENNZApi();
+	const { selectedWallet } = useWalletProvider();
+
+	const selectedAccount = useSelectedAccount();
+
+	return useCallback(
+		async (setBalances) => {
```

Comment:
just realise you have `setBalances` to be passed on both the hook and the callback, is there a reason for it?